### PR TITLE
AB#167769: Website redesign

### PR DESCRIPTION
**Pull request by JoeHeeley2**
# Overview

- Cosmetic changes to the front end. 
- Added login to landing page
- Added quick access panel to home page
- updated bootstrap version for icons

## Azure Boards

- [AB#167769](https://dev.azure.com/UniversityOfNottingham/bd2de5ba-2a41-4bdc-a444-61e6b706132a/_workitems/edit/167769)
- [AB#167775](https://dev.azure.com/UniversityOfNottingham/bd2de5ba-2a41-4bdc-a444-61e6b706132a/_workitems/edit/167775)

### CODE REVIEW

File: Webapp/sources/blueprints/main/routes.py [hunk 1]
```diff
@@ -33,38 +34,34 @@ def index() -> Response:
...
-            db.session.query(models.User)
-            .filter(models.User.email == current_user.email)
-            .first()
-        )
-        user_confirmed = user.is_verified
-        user_role = user.Role.name
+        form = None
+        user_role = current_user.Role.name
+        user_confirmed = current_user.is_verified
         workgroups = get_workgroups()
```
Comment (JoeDavies-6) [inline Webapp/sources/blueprints/main/routes.py:56]: I don't think we need the workgroups/notification number here if its just for the navbar. The inject_session_context in sources/_init__.py should do these for all pages now.

Comment (JoeHeeley2) [reply Webapp/sources/blueprints/main/routes.py]: Removed workgroups and notification number from here now. Notification number was missing from inject_session_context so I've added it

File: Webapp/sources/blueprints/main/routes.py [hunk 2]
```diff
@@ -33,49 +34,71 @@ def index() -> Response:
...
+        icon_names = [i.name for i in workgroups]
+        header = "Workbooks in " + selected
+        load_type = "workbook"
+        bootstrap_icon = "bi bi-journal-text"
+
+    elif load_type == "workbook":
+        reactions = services.reaction.list_active_in_workbook(
+            workbook=selected, workgroup=request.json.get("activeWorkgroup"), sort_crit="time"
+        )
+        icon_names = [i.reaction_id for i in reactions[:11]]
```
Comment (psytb6) [inline Webapp/sources/blueprints/main/routes.py:92]: Potential error if receiving no reactions or workbooks

File: Webapp/sources/blueprints/main/routes.py [hunk 3]
```diff
@@ -33,49 +34,71 @@ def index() -> Response:
...
+    """
+    This function renders the icon macro from macros.html for the quick access panel
+    """
+    selected = request.json.get("input")
+    load_type = request.json.get("load_type")
+    icon_names = []
+    header = None
+    bootstrap_icon = ""
+    if load_type == "workgroup":
+        workgroups = services.workbook.get_workbooks_from_user_group_combination(selected)
```
Comment (psytb6) [inline Webapp/sources/blueprints/main/routes.py:82]: variable name 'workgroups' calls 'get_workbook'

**Review by JoeDavies-6 — APPROVED** [review]

File: Webapp/sources/blueprints/main/routes.py (hunk 2 above)
Comment (JoeHeeley2) [reply Webapp/sources/blueprints/main/routes.py:92]: If users have no workbooks/reactions then the functions just return empty lists, so icon_lists is empty and no icons are rendered. This doesnt cause an error and you can create a new workbook/reaction in the access panel. is that what you meant?

Comment (JoeDavies-6) [reply Webapp/sources/blueprints/main/routes.py:92]: that sounds fine then. I thought line 92 may cause an error if reactions was None

Comment (JoeHeeley2) [reply Webapp/sources/blueprints/main/routes.py:92]: It probably would, I'll add a catch just to be safe

File: Webapp/sources/static/css/pagestyle.css [hunk 4]
```diff
@@ -69,6 +69,16 @@ body {
   border-color: #686868;
 }
 
+.btn.btn-outline-primary {
+  border-color: #777778;
```
[the remark this answers is not in the document]
Comment (JoeHeeley2) [reply Webapp/sources/static/css/pagestyle.css]: added variables for repeating colours in colour pallette

File: Webapp/sources/blueprints/main/routes.py (hunk 3 above)
Comment (JoeHeeley2) [reply Webapp/sources/blueprints/main/routes.py]: changed the variable name

**Review by JoeDavies-6 — APPROVED** [review]

**Review by psytb6 — APPROVED** [review]

**Pull request merged**